oap-stdlib/src/main/java/oap/http/server/nio/NioHttpServer.java

```diff
@@ -211,20 +210,18 @@ private synchronized SSLContext getSSLContext() {
         try( InputStream inputStream = Files.newInputStream( sslConfiguration.jks ) ) {
             KeyStore ks = KeyStore.getInstance( "JKS" );
             KeyManagerFactory kmf = KeyManagerFactory.getInstance( KeyManagerFactory.getDefaultAlgorithm() );
-            sslContext = SSLContext.getInstance( "TLS" );
 
             ks.load( inputStream, sslConfiguration.password.toCharArray() );
             kmf.init( ks, sslConfiguration.password.toCharArray() );
 
-            sslContext.init( kmf.getKeyManagers(), null, null );
+            keyManagers = kmf.getKeyManagers();
         } catch( IOException e ) {
             throw new UncheckedIOException( e );
-        } catch( UnrecoverableKeyException | CertificateException | KeyStoreException | NoSuchAlgorithmException |
-                 KeyManagementException e ) {
+        } catch( UnrecoverableKeyException | CertificateException | KeyStoreException | NoSuchAlgorithmException e ) {
             throw new RuntimeException( e );
```

Comment:
do we really want to throw a RuntimeException? If the http server is configured to do both HTTP and HTTPS, it should do HTTP even if the HTTPS setup fails. NioHttpServer though only throw a RuntimeException if HTTPS setup fails and it is not configured to do HTTP.

Reply:
It's strange when we configure http and https and the server starts with a missing certificate for example.

Reply:
I agree that the best answer is not obvious. I feel however that if you ask the code to do HTTP and HTTPS but it cannot do HTTPS, then if should log an error and do HTTP while I fix the problem with the keystore. My use case is that I'm adding HTTPS to a system currently working with just HTTP,  and I'd like to be sure not to break HTTP while I'm adding HTTPS.

Reply:
We test the **https** implementation with tests and it either works or the tests fail. The only thing we don't test with unit tests is the working configuration, but that's not the case with this code.

Reply:
as you wish, however your tests will not stop the user from providing a bad (or missing) keystore.jks. In this case throwing a runtimexception will break a valid HTTP configuration if HTTPS is also requested